akka-stream/src/main/scala/akka/stream/impl/fusing/StreamOfStreams.scala

```diff
@@ -269,7 +269,7 @@ final class GroupBy[T, K](maxSubstreams: Int, keyFor: T ⇒ K) extends GraphStag
               subSubstreamSource.push(nextElementValue)
               clearNextElement()
             }
-          } else tryPull(in)
+          } else if (!hasBeenPulled(in)) tryPull(in)
```

Comment:
hm? That's not what we're saying.
`tryPull` works, `pull` works. `tryPull` does additional `isClosed` checks, which we don't need to do, because we guard the pull already with the `hasBeenPulled`